WordPress/gutenberg
# [RNMobile] Reusable blocks - Present block content (preview mode)

`Gutenberg-mobile PR` -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/2953

**Second part (adding Reusable blocks to the inserter menu):**
`Gutenberg PR` -> https://github.com/WordPress/gutenberg/pull/25383
`Gutenberg-mobile PR` -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/2644

## Description
This PR is the first part of the issue [Add Support to Insert Existing Reusable Blocks from the Editor](https://github.com/wordpress-mobile/gutenberg-mobile/issues/2575).

The scope of this issue is limited to add Reusable blocks to the inserter menu, however due to the need of registering the Reusable block type (`core/block`), I also had to add support for presenting it in the editor.

The edition support will be added in this [issue](https://github.com/wordpress-mobile/gutenberg-mobile/issues/2597), so for now this PR only provides the preview mode of this type of block.

## How has this been tested?
I tested this feature in the following 3 different scenarios:

<details>
<summary><strong>WordPress.com site</strong></summary>

Create a WordPress.com site or use an already created one.
- Create a post in the web version
- Add some blocks and create a Reusable block
- Open the app with metro running (Gutenberg-mobile)
- Add the WordPress.com site
- Edit the previous created post
- Check that the Reusable block is shown
- Tap on it and check that the title of the Reusable blocks is shown

</details>

<details>
<summary><strong>Self-hosted site with Jetpack</strong></summary>

Create a self-hosted site with Jetpack (I used https://jurassic.ninja/create/) or use an already created one.
- Create a post in the web version
- Add some blocks and create a Reusable block
- Open the app with metro running (Gutenberg-mobile)
- Add the self-hosted site
- Edit the previous created post
- Check that the Reusable block is shown
- Tap on it and check that the title of the Reusable blocks is shown

</details>

<details>
<summary><strong>Self-hosted site without Jetpack</strong></summary>

Create a self-hosted site without Jetpack (I created a local one using [this instructions](https://developer.wordpress.org/block-editor/contributors/develop/getting-started/#local-environment)) or use your own.
- Create a post in the web version
- Add some blocks and create a Reusable block
- Open the app with metro running (Gutenberg-mobile)
- Add the self-hosted site
- Edit the previous created post
- Check that the Reusable block is shown
- Tap on it and check that the title of the Reusable blocks is shown

</details>

## Screenshots 
<details>
<summary>iOS Screenshots</summary>


## iPhone 8

![ios-reusable-block](https://user-images.githubusercontent.com/14905380/94845321-d0e38100-041f-11eb-93e4-63efb1c34cd8.gif)


Reusable block unselected | Reusable block selected
------------ | -------------
<img src=https://user-images.githubusercontent.com/14905380/94845326-d214ae00-041f-11eb-8b57-6402e72d7731.png width=380> | <img src=https://user-images.githubusercontent.com/14905380/94845330-d345db00-041f-11eb-85fd-40624e846ffe.png width=400>


</details>

<details>
<summary>Android Screenshots</summary>

## Nexus 5

![android-reusable-block](https://user-images.githubusercontent.com/14905380/94844593-c4125d80-041e-11eb-839d-8a7920b353c9.gif)

Reusable block unselected | Reusable block selected
------------ | -------------
<img src=https://user-images.githubusercontent.com/14905380/94844599-c70d4e00-041e-11eb-8bb6-4966f44a817b.png width=390> | <img src=https://user-images.githubusercontent.com/14905380/94844601-c83e7b00-041e-11eb-8a5b-1b8f25f971fc.png width=380>

</details>

<details>
<summary>Unsupported Block Editor support - Self-hosted site</summary>

Without Jetpack | With Jetpack
------------ | -------------
<img src=https://user-images.githubusercontent.com/14905380/104010176-67a4a400-51ac-11eb-8023-5c3ff5a3260b.png width=390> | <img src=https://user-images.githubusercontent.com/14905380/104010179-68d5d100-51ac-11eb-9975-5c2a8685c3b3.png width=380>

</details>

## Types of changes
New feature

## Checklist:
